File: vigil-backend/app/schemas/repository.py
from datetime import datetime
from typing import List, Optional
import uuid
from pydantic import BaseModel, ConfigDict, Field
from app.schemas.base import PaginatedResponse


class RepositoryRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID
    github_repo_id: int
    owner_login: str
    name: str
    full_name: str
    default_branch: str
    private: bool
    html_url: str
    is_active: bool
    created_at: datetime
    updated_at: datetime


class RepositoryListResponse(PaginatedResponse):
    items: List[RepositoryRead]

File: vigil-frontend/src/types/repository.ts
import type { PaginatedResponse } from './api';

export interface RepositoryRead {
  id: string; // uuid
  user_id: string; // uuid
  github_repo_id: number;
  owner_login: string;
  name: string;
  full_name: string;
  default_branch: string;
  private: boolean;
  html_url: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export type RepositoryListResponse = PaginatedResponse<RepositoryRead>;
